### ISSUE TITLE
Pact file is not parsed correct starting with version 4.1.2 when using XML requests in body

### ISSUE THREAD

**Issue by anddreiu** [DiUS/pact-jvm#1218]
After upgrading the pact provider junit library from 4.1.1 to 4.1.2 there are some issues when running the provider tests because the Pact file is not parsed.

```
au.com.dius.pact.core.support.json.JsonException: Invalid JSON (1:2), found unexpected character '<'
	at au.com.dius.pact.core.support.json.JsonLexer.unexpectedCharacter(JsonParser.kt:88)
	at au.com.dius.pact.core.support.json.JsonLexer.nextToken(JsonParser.kt:79)
	at au.com.dius.pact.core.support.json.JsonParser.nextTokenOrThrow(JsonParser.kt:278)
	at au.com.dius.pact.core.support.json.JsonParser.parse(JsonParser.kt:151)
	at au.com.dius.pact.core.support.json.JsonParser.parseString(JsonParser.kt:131)
	at au.com.dius.pact.core.model.RequestResponseInteraction$Companion.setupBodyForJson(RequestResponseInteraction.kt:123)
	at au.com.dius.pact.core.model.RequestResponseInteraction$Companion.requestToMap(RequestResponseInteraction.kt:85)
	at au.com.dius.pact.core.model.RequestResponseInteraction.toMap(RequestResponseInteraction.kt:37)
	at au.com.dius.pact.provider.reporters.JsonReporter.interactionDescription(JsonReporter.kt:132)
	at au.com.dius.pact.provider.ProviderVerifier.reportInteractionDescription(ProviderVerifier.kt:502)
	at au.com.dius.pact.provider.junit.InteractionRunner.interactionBlock(InteractionRunner.kt:218)
	at au.com.dius.pact.provider.junit.InteractionRunner.run(InteractionRunner.kt:162)
	at au.com.dius.pact.provider.junit.PactRunner.runChild(PactRunner.kt:150)
	at au.com.dius.pact.provider.junit.PactRunner.runChild(PactRunner.kt:56)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
```

I've noticed that the problem appears when the Content-type for the interaction request is not specified:

```
"headers": {
        "Content-Type": "text/xml"
}
```

With the previous versions it was not mandatory to add this.

Thanks,
Andrei

**Comment by uglyog** [comment]
4.1.9 released with the fix

**Comment by anddreiu** [comment]
I can confirm that the issue is fixed in 4.1.9 version. Thank you.
